### DISCUSSION TITLE
Collections within collections - are they possible?

### DISCUSSION THREAD

**Discussion by rachaelfurnbullard** [11ty/eleventy#1735] [Q&A] [answered]
I'm struggling to figure something out and I'm here asking for help on whether it's possible, or whether I'm trying to be too adventurous. 

I followed this tutorial on creating pages from data: https://www.11ty.dev/docs/pages-from-data/

My data is in /_data/venues.json and it happily generates venue1/index.html, venue2.html, venue3/index.html, and so on. Everything is great so far.

My question is... How/Can I then output data from another collection within those generated files?

I have /_data/venue/venue1.json, /_data/venue/venue2.json, /_data/venue/venue3.json and I'm (trying) using nunjucks. 

In my venue-page.njk which generates the html pages, I have this:

```
{% for item in venue.venue1 %}
// output
{% endfor %}
```

and it doesn't work.

I've also tried:

```
{% for item in venue/venue1 %}
// output
{% endfor %}
```

and unfortunately this doesn't work either. 

I'm hoping this will be a simple answer for someone who knows a lot more about 11ty / nunjucks / json than I do. Is what I'm trying to do even possible? And if so, how might I go about doing it? Thank you so much for any help / insight you can offer, even if it's just 'ahahaha no this will not work.'

**Comment by pdehaan** [comment, marked as answer]
I definitely think it's possible, but might be a bit tricky based on how your folders and variables are named.

Here's my _data/venues.json file:

```json
[
  { "name": "Venue 1", "key": "venue1" },
  { "name": "Venue 2", "key": "venue2" }
]
```

And my _data/venue/venue1.json (and venue2.json) files look roughly like this:

```json
{
  "name": "Venue 1",
  "address": "100 Main St, City, State, Country"
}
```

This is where is starts to get a bit tricky...
Here's my venue-page.njk template:

```njk
---
title: Venues
pagination:
  data: venues
  size: 1
  alias: _venue
permalink: "/{{ _venue.name | slug }}/"
---

{# `venue` here refers to /_data/venue/* directory. #}
{% set $venue = venue[_venue.key] %}

<p>name={{ $venue.name }}</p>
<p>address={{ $venue.address }}</p>
<p>venues={{ venues | dump }}</p>
```

First we're looping over the `venues.json` global data file, assigning an `alias` of `_venue` (which we can use for the dynamic `permalink`, etc). If I choose an alias of `venue` without the underscore, it seems to conflict with my _data/venue/venue1.json file since I can no longer use `venue` to access that global data file. But, by using a different alias, I can still access the current loop value, and the global data file, and make an even more confusing third reference to `$venue` which is a reference to the _data/venue file key defined in the _data/venues.json `key` property (if any of that made sense).

I'm happy to push my sample repo up to GitHub if you want to take a look at my workaround, or I can help troubleshoot your code if you have a public repo.

**Reply by rachaelfurnbullard** [reply]
Oh thank you so much! I'll go and meddle with my code and see what I can convince to work. 

Fortunately in my actual code they aren't called venue1 and venue2, they're called things like Albert Park and Indianapolis, but I thought I'd leave those out of my example so as not to cause confusion, but it seems like I might have caused more confusion!

My repo is currently private but if I can find the way to hide my API keys and stuff then I'll make it public.

Thank you! <3

**Reply by rachaelfurnbullard** [reply]
OHMYGOD I figured it out. 

```
{% set $venue = venue[_venue.fields.Slug] %}

{% for item in $venue %}
<p>Series Friendly Name: {{ item.fields.SeriesFriendlyName }}</p>
<p>Official Name: {{ item.fields.OfficialName }}</p>

{% endfor %}
```

Thank you SO MUCH for your help, kind stranger on the internet, Peter from Canada. I wish you and your family the best of health and success.
